pkg/iter/entry_iterator.go

```diff
@@ -246,7 +233,7 @@ Outer:
 				!entry.Timestamp.Equal(i.buffer[0].Entry.Timestamp) {
 				break
 			}
-			for _, t := range previous {
+			for _, t := range i.buffer {
```

Comment:
Entries may have been added to the buffer since `previous` was created so we need to iterate the actual buffer on each subsequent entry.